src/Http/Routing/src/RouteEndpointDataSource.cs

```diff
@@ -113,6 +113,8 @@ internal RouteEndpointBuilder GetSingleRouteEndpointBuilder()
 
     [UnconditionalSuppressMessage("Trimmer", "IL2026",
         Justification = "We surface a RequireUnreferencedCode in the call to the Map method adding this EndpointDataSource. The trimmer is unable to infer this.")]
+    [UnconditionalSuppressMessage("AOT", "IL3050",
+        Justification = "We surface a RequiresDynamicCode in the call to the Map method adding this EndpointDataSource. The trimmer is unable to infer this.")]
```

Comment:
Is this valid? I see a path that creates a `RouteEndpointDataSource` that isn't annotated.

https://github.com/dotnet/aspnetcore/blob/53124ab8cbf152f59120982f1c74e802e7970845/src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs#L145-L153

https://github.com/dotnet/aspnetcore/blob/53124ab8cbf152f59120982f1c74e802e7970845/src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs#L188-L199

https://github.com/dotnet/aspnetcore/blob/53124ab8cbf152f59120982f1c74e802e7970845/src/Http/Routing/src/Builder/EndpointRouteBuilderExtensions.cs#L410-L431

Reply:
This is fine. Map calls with `RequestDelegate` (instead of `Delegate`) are safe.

Reply:
We should update the justification then as it is not correct.

Reply:
Looking at the method, it looks like it will call into RequestDelegateFactory.Create even if it is a `RequestDelegate` when there are FilterFactories on the builder:

https://github.com/dotnet/aspnetcore/blob/53124ab8cbf152f59120982f1c74e802e7970845/src/Http/Routing/src/RouteEndpointDataSource.cs#L224-L230

In this case, are we sure it is still AOT safe?

Reply:
> Looking at the method, it looks like it will call into RequestDelegateFactory.Create even if it is a `RequestDelegate` when there are FilterFactories on the builder:
> 
> https://github.com/dotnet/aspnetcore/blob/53124ab8cbf152f59120982f1c74e802e7970845/src/Http/Routing/src/RouteEndpointDataSource.cs#L224-L230
> 
> In this case, are we sure it is still AOT safe?

Yes. The if statement will be false for `RequestDelegate` methods. TBH this type does too much.

Reply:
I updated the suppression justification to discuss `RequestDelegate` map methods.

Reply:
> The if statement will be false for RequestDelegate methods.

Even when there are FilterFactories? There’s an OR in that if statement.

Reply:
Can someone remind me what is not trim and AOT compatible about RequestDelegateFactory? Is it just the JSON (de)serialization? Or is there something else? The use of Linq.Expressions?